File: VB/WindowsFormsApplication1/MyDockManager/MyDockManager.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraBars.Docking.Paint
Imports DevExpress.XtraBars.Styles
Imports DevExpress.Utils
Imports DevExpress.XtraBars.Docking

Namespace WindowsFormsApplication1
	<System.ComponentModel.DesignerCategory("")> _
	Public Class MyDockManager
		Inherits DockManager

		Public Sub New()

		End Sub
		Public Sub New(ByVal container As IContainer)
			MyBase.New(container)

		End Sub
		Public Sub New(ByVal form As ContainerControl)
			MyBase.New(form)

		End Sub

		Protected Overrides Function CreateDockPanel(ByVal dock As DockingStyle, ByVal createControlContainer As Boolean) As DockPanel
			Return New MyDockPanel(createControlContainer, dock, Me)
		End Function


		Private Shared ReadOnly customDraw As Object = New Object()
		Public Custom Event CustomDrawPanelCaption As CustomDrawEventHandler
			AddHandler(ByVal value As CustomDrawEventHandler)
				Me.Events.AddHandler(customDraw, value)
			End AddHandler
			RemoveHandler(ByVal value As CustomDrawEventHandler)
				Me.Events.RemoveHandler(customDraw, value)
			End RemoveHandler
			RaiseEvent(ByVal sender As Object, ByVal e As MyCustomDrawArgs)
			End RaiseEvent
		End Event


		Protected Friend Overridable Sub RaiseCustomDraw(ByVal args As MyCustomDrawArgs)
			Dim handler As CustomDrawEventHandler = CType(Me.Events(customDraw), CustomDrawEventHandler)
			If handler IsNot Nothing Then
				handler(Me, args)
			End If
		End Sub

	End Class
End Namespace

File: CS/WindowsFormsApplication1/MyDockManager/MyDockManager.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraBars.Docking.Paint;
using DevExpress.XtraBars.Styles;
using DevExpress.Utils;
using DevExpress.XtraBars.Docking;

namespace WindowsFormsApplication1
{
    [System.ComponentModel.DesignerCategory("")]
    public class MyDockManager : DockManager
    {

        public MyDockManager()
        {

        }
        public MyDockManager(IContainer container)
            : base(container)
        {

        }
        public MyDockManager(ContainerControl form)
            : base(form)
        {

        }

        protected override DockPanel CreateDockPanel(DockingStyle dock, bool createControlContainer)
        {
            return new MyDockPanel(createControlContainer, dock, this);
        }


        static readonly object customDraw = new object();
        public event CustomDrawEventHandler CustomDrawPanelCaption
        {
            add { this.Events.AddHandler(customDraw, value); }
            remove { this.Events.RemoveHandler(customDraw, value); }
        }


        protected internal virtual void RaiseCustomDraw(MyCustomDrawArgs args)
        {
            CustomDrawEventHandler handler = (CustomDrawEventHandler)this.Events[customDraw];
            if (handler != null) handler(this, args);
        }

    }
}